### PR TITLE
feat: new scc code to retrieve latest scc profile version

### PR DESCRIPTION
**Pull request by jor2**
### Description

The logic [here](https://github.com/terraform-ibm-modules/terraform-ibm-scc/blob/13e104e88ca1790b171431d434fdc05a5af7d695/modules/attachment/main.tf#L19-L25) assumes that each profile name will always have a latest = true value. However we have seen the SCC team change the profile names, meaning we can no longer rely on that logic.

Implementing our own logic to determine the latest version of a profile.

lookup all versions of a profile
sort them
return the latest
That way we ensure there is ALWAYS a latest version for every profile name
### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

update SCC logic around determining latest profile version

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: modules/attachment/main.tf [hunk 1]
```diff
@@ -16,9 +16,30 @@ data "ibm_scc_profiles" "scc_profiles" {
 }
 
 locals {
+  # Get profiles by name matching that provided in var.profile_name
+  relevant_profiles = [
```
Comment (ocofaigh) [inline modules/attachment/main.tf:20]: trying to follow the logic... this should return 1 profile right? so maybe rename it to `relevant_profile`

File: modules/attachment/outputs.tf [hunk 2]
```diff
@@ -7,3 +7,13 @@ output "attachment_parameters" {
...
+
+output "sorted_list" {
+  description = "SCC profile attachment parameters"
+  value       = local.sorted_list
+}
+
+output "latest_profile" {
+  description = "SCC profile attachment parameters"
+  value       = local.latest_profile
+}
```
Comment (ocofaigh) [inline modules/attachment/outputs.tf:19]: I don't think either of these are needed as outputs. All a user will care about is maybe the profile name and version used by the attachment

**Review by ocofaigh — CHANGES_REQUESTED** [review]
see comments

File: modules/attachment/main.tf (hunk 1 above)
Comment (jor2) [reply modules/attachment/main.tf:20]: No, it returns all the profile versions for the listed profile such as "IBM Cloud Financial Services". We then take that list and recreate the list from lowest to latest versions and the latest version will be the last in the list so we just index the last element and take that profile.

Comment (ocofaigh) [reply modules/attachment/main.tf:20]: it returns all profiles for that 1 profile thought right? So all versions with the profile name "IBM Cloud Financial Services" ?

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**